### DISCUSSION TITLE
Revalidate both first & last page in useSWRInfinite

### DISCUSSION THREAD

**Discussion by arunoda** [vercel/swr#616] [Ideas]
Currently `useSWRInfinite` only revalidate the first page. This is okay when we sort pages by **latest to oldest**. If it's the other way around `useSWRInfinite` only revalidate for a page which is not going to change. (most of the time)

I think it'll be great, if there's an option to revalidate the last page & the first page. Here's I'm talking about the last page of the whole list. Not the last page at a given time.

If that's the case, maybe we can revalidate it automatically without passing any option.

**Comment by shuding** [comment]
> I think it'll be great, if there's an option to revalidate the last page & the first page.

This sounds like a good idea. Or, we can even provide an option like `shouldRevalidatePage: index => boolean`.

**Reply by svenlombaert** [reply]
Any updates on this idea? I would actually love to have the control to revalidate specific pages when using useSWRInfinite. Doesn't seem possible currently? 

eg.: I have three pages in cache. It revalidates the first page on mount, but there's no option to then revalidate the second page? Unless I revalidate all, but that's not what I want, cause a user could have loaded 10 pages and then we'd do 10 requests.

My use case for this is, that upon re-entering a certain screen, I visually limit the loaded pages to the first one and have a 'see more' button, which either loads the next page (if not in cache) or just renders the next items if the page is already in cache. For the latter, I'd also like to revalidate that page.

**Reply by carlfranciscastillo** [reply]
Maybe this should be added to the bound mutate rather than being an option? Since it would be nice if I mutated something in my paginated list then SWR would only revalidate that page/index. I agree that revalidating all is too costly and wasteful and goes against the idea of pagination in the first place

**Reply by Kamleshpaul** [reply]
any update on this i also need this feature to load chat messages when someone send or receive  chat i need to revalidate last page which is not possible seems currently

**Comment by koba04** [comment]
I just experimented with the `shouldRevalidatePage` option like this.
https://github.com/koba04/swr/commit/c952111e801457d1d025b775ae6648073e659f34

But there are some considerations for the option.

- how to behave with `revalidateFirstPage` and `revalidateAll`
- When `shouldRevalidatePage` returns false, but the page data is stale or empty, should we revalidate it or not?

There's also another request related to this. This is a feature request to revalidate a specific page like `mutatePage()`, so it's not the same, but we have to recognize it to find out the solution.
https://github.com/vercel/swr/discussions/2271

**Comment by BartoszTroszka** [comment]
Could we possibly get that feature? There is still a need for that.

**Comment by koba04** [comment]
I've added a feature to revalidate specific pages by passing a function to the `revalidate` option. Would you be willing to try it?
https://github.com/vercel/swr/pull/2862
